lib/transports/websocket.js

```
@@ -74,10 +74,12 @@
     };
     this.websocket.onclose = function () {
       self.onClose();
+      self.socket.onFailed();
```

This allows WebSockets to fail faster instead of waiting for a connect timeout. For example if a new spec is released and the server doesn't supports it yet...